src/app/(views)/(public)/c/[catalogId]/add-to-fav.tsx

```diff
@@ -1,57 +1,42 @@
 "use client";
 
+import { useLiveQuery } from "dexie-react-hooks";
 import { useEffect, useState } from "react";
 
 import { StarIcon } from "~/components/custom/icons";
 import JustTip from "~/components/custom/just-the-tip";
 import { Button } from "~/components/shadcn/button";
 import { toast } from "~/hooks/use-toast";
+import { db } from "~/utils/db";
 
 export const AddToFavorites = ({
   catalogId,
-  catalogDescription,
   catalogTitle,
+  catalogDescription,
 }: any) => {
-  const [existingCatalogs, setExistingCatalogs] = useState<any[]>([]);
-
-  useEffect(() => {
-    setExistingCatalogs(
-      JSON.parse(window?.localStorage?.getItem("favorites") || "[]")
-    );
-  }, []);
-
+  let favoriteCatalogs = useLiveQuery(() => db["favorites"].toArray(), []) ?? []
   const [catalogExists, setCatalogExists] = useState<boolean>(false);
 
   useEffect(() => {
     checkIfExists();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [existingCatalogs]);
+  }, [favoriteCatalogs]);
 
   const checkIfExists = () => {
-    for (let i = 0; i < existingCatalogs?.length; i++) {
-      if (existingCatalogs[i].id === catalogId) {
+    for (let i = 0; i < favoriteCatalogs?.length; i++) {
+      if (favoriteCatalogs[i].id === catalogId) {
         setCatalogExists(true);
         return;
       }
     }
     setCatalogExists(false);
   };
 
-  const addToFav = () => {
+  const addToFav = async () => {
     if (catalogExists) {
-      // Remove the catalogId
-      const filterCatalogIds = existingCatalogs.filter(
-        (item: any) => item.id != catalogId
-      );
-
       toast({ title: "Catalog removed from favorites." });
 
-      window?.localStorage?.setItem(
-        "favorites",
-        JSON.stringify(filterCatalogIds)
-      );
-
-      setExistingCatalogs(filterCatalogIds);
+      await db["favorites"].delete(catalogId);
```

Comment:
I used it like this because, when I did the 'watch later' part, I named the database 'watch-later,' separated by a dash. So, I had to use square brackets to access it. Since I used it in one case, and if the word is separated by a dash, I can't access it without square brackets, I used them just as a convention.

Reply:
It makes sense. 👍🏻